lib/eventboss/error_handlers/sentry.rb

```diff
@@ -0,0 +1,19 @@
+module Eventboss
+  module ErrorHandlers
+    class Sentry
+      def call(exception, context = {})
+        eventboss_context = { component: 'eventboss' }
+        eventboss_context[:action] = context[:processor].class.to_s if context[:processor]
+
+        default_options = { }
+
+        ::Sentry.with_scope do |scope|
```

Comment:
If we use `Sentry` in our gem then we should probably mark this as a project's dependency (`eventboss.gemspec`)

Reply:
I think this still is a dependency on the gem - it uses Sentry API that is valid for the specific range of Sentry versions

Reply:
Yeah, but what if I use Rollbar, not a Sentry? My whole project would have Sentry downloaded and required when then. As this is plugin, we don't force you to that feature. Unfortunately, as gem creators, we don't have options for checking the dependency, so you should take care about it by yourself. In that case you probably have it in your decencies, because you have to have it configured. Only version validation is still missing.

Reply:
I see this that way - Sentry is clearly dependency here. Its compatibility needs to be ensured somewhere - either in the client app  or the gem side.

I see no harm that user needs to download all the gems related with our plugins if we decided to keep them within the repo - it's still better that leave things unsecured in hope that nothing will blow up :)  

We can still create some other gems like `eventboss-extensions` where all the plugins would be downloaded if the user really wants to. We could provide event greater granularity by exposing things like`eventboss-rollbar` etc.

The third option would be to inform users that there's a hook for custom error handlers so that you can attach one if you want - in respect to your Gemfile.

Reply:
I think this is not a place for this `Eventboss::ErrorHandlers::Sentry`:
* This class introduces external dependancy `Sentry` and it's unnecessary:
What if I'm using rollbar? Then I wouldn't need to load additional gem.
* Scenario when I'm having different implementation and I don't want to use this one - unnecessary require.

What I would suggest is to just simply pass lambda to `error_handlers` in `Eventboss` initializer:
```
Eventboss.configure do |config|
  config.eventboss_app_name = ['EVENTBUS_APP_NAME'
  config.logger = Rails.logger
  config.error_handlers << lambda do |exception, context = {}|
    # code
  end
  config.log_level = Rails.configuration.log_level
end

```
And dependency to `Sentry` only lives in your application

Reply:
I think that anyway by default these files from `error_handlers` are not loaded. So far nor Rollbar neither AirBrake gems were not marked as eventboss dependency. 

As `eventboss` user I would like to delegate anything to this lib. With lambda approach every client app would need to get dig into what can be in `context` which basically requires digging deep into eventboss, sqs etc.

Reply:
> As eventboss user I would like to delegate anything to this lib. With lambda approach every client app would need to get dig into what can be in context which basically requires digging deep into eventboss, sqs etc.
That's fair point! I think we can proceed like this 👍